### ISSUE TITLE
How to apply the same transformation for test and train datasets

### ISSUE THREAD

**Issue by eyadsibai** [spotify/featran#14]
It's not clear to me how would you apply the transformation to the train dataset then to the test dataset.

It is important to save the state of the transformation after applying it to the train dataset. So the same values is being used for the test.

**Comment by eyadsibai** [comment]
for example the minmax scaler it is not stateless transformation

**Comment by richwhitjr** [comment]
There is a settings that can be saved after the aggregation.  When running on a test set you can use extractWithSettings that uses the aggregations from the training set.  In this case the transformations won't be recomputed but instead applied from the original training run.

From the Example:

```scala
// Extract settings as a JSON string
val settings = f1.featureSettings

// Extract features from new records, but reuse previously saved settings
val f2 = spec.extractWithSettings(records.take(records.size / 2), settings)
```

**Comment by eyadsibai** [comment]
thanks for the explanation!
